lib/arsi/arel_tree_manager.rb

```diff
@@ -4,23 +4,25 @@ module Arsi
   module ArelTreeManager
     # This is from Arel::SelectManager which inherits from Arel::TreeManager.
     # We need where_sql on both Arel::UpdateManager and Arel::DeleteManager so we add it to the parent class.
-    if ::Arel::VERSION[0].to_i >= 6
-      def where_sql(engine = Arel::Table.engine)
-        return if @ctx.wheres.empty?
+    def where_sql(provided_engine = :none)
+      return if @ctx.wheres.empty?
 
-        viz = if ::Arel::VERSION[0].to_i > 6
-          ::Arel::Visitors::WhereSql.new(engine.connection.visitor, engine.connection)
+      selected_engine = provided_engine
+      if selected_engine == :none
+        # Arel 9 has no engine reader
+        selected_engine = if self.respond_to?(:engine)
+          self.engine || Arel::Table.engine
         else
-          ::Arel::Visitors::WhereSql.new engine.connection
+          Arel::Table.engine
         end
-        ::Arel::Nodes::SqlLiteral.new viz.accept(@ctx, ::Arel::Collectors::SQLString.new).value
       end
-    else
-      def where_sql
-        return if @ctx.wheres.empty?
-        viz = ::Arel::Visitors::WhereSql.new @engine.connection
-        ::Arel::Nodes::SqlLiteral.new viz.accept @ctx
```

Comment:
Are we dropping support for < 6?

Reply:
We already don't test for it. Rails 4 is on 6